src/Compilers/CSharp/Portable/Lowering/SpillSequenceSpiller.cs

```diff
@@ -469,6 +468,26 @@ private BoundStatement UpdateStatement(BoundSpillSequenceBuilder builder, BoundS
                                                call.Method,
                                                ImmutableArray.Create(Spill(builder, call.Arguments[0]), Spill(builder, call.Arguments[1])));
                         }
+                        else if (call.Method == _F.Compilation.GetWellKnownTypeMember(WellKnownMember.System_String__op_Implicit_ToReadOnlySpanOfChar))
+                        {
+                            Debug.Assert(call.Arguments.Length == 1);
+                            return call.Update([Spill(builder, call.Arguments[0])]);
+                        }
+
+                        goto default;
+
+                    case BoundKind.ObjectCreationExpression:
+                        var objectCreationExpression = (BoundObjectCreationExpression)expression;
+
+                        if (refKind == RefKind.None &&
+                            objectCreationExpression.InitializerExpressionOpt is null &&
+                            objectCreationExpression.Constructor.OriginalDefinition == _F.Compilation.GetWellKnownTypeMember(WellKnownMember.System_ReadOnlySpan_T__ctor_Reference))
+                        {
+                            Debug.Assert(objectCreationExpression.Arguments.Length == 1);
+                            return objectCreationExpression.UpdateArgumentsAndInitializer([Spill(builder, objectCreationExpression.Arguments[0], objectCreationExpression.ArgumentRefKindsOpt[0])],
```

Comment:
Is it even possible for `ArgumentRefKindsOpt` to be `default` at this phaze of compilation? Since the first argument of constructor is passed by reference, `ArgumentRefKindsOpt` can be `default` only in error scenario. So is it possible to have an error here or we just want to be extra safe?

Reply:
> Is it even possible for ArgumentRefKindsOpt to be default at this phaze of compilation? 

Since theoretically we can run into a node that doesn't originate from string concat optimization (could be even a node originated from source), if not now, then in the future. It doesn't hurt to be extra careful. In my opinion, it is much simple to write defensive code than to get confident that the answer to the question is "No" for all possible situations